pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -386,7 +386,15 @@ public Map<String, TaskPartitionState> getSubtaskStates(
   @ApiOperation("Get the task config (a list of child task configs) for the given task")
   public List<PinotTaskConfig> getTaskConfigs(
       @ApiParam(value = "Task name", required = true) @PathParam("taskName") String taskName) {
-    return _pinotHelixTaskResourceManager.getTaskConfigs(taskName);
+    return _pinotHelixTaskResourceManager.getSubtaskConfigs(taskName);
+  }
+
+  @GET
+  @Path("/tasks/task/{taskName}/runtime/config")
+  @ApiOperation("Get the task runtime config for the given task")
+  public Map<String, String> getTaskConfig(
```

Comment:
just another thought: is it better to add a separate endpoint or is it better to include more information in the existing endpoint?

Reply:
I'd prefer a separate API, in case any applications are assuming a List<PinotTaskConfig> returned by the old API. (if we have versioned our APIs with prefix like v1/v2, I'd save a lot of time to come up with a new name :p)